pages/doc/integrations_tas_custom_app_metrics_howto.md

```diff
@@ -0,0 +1,75 @@
+
+To instrument an application for metrics registrar scraping, you will typically use either a prometheus SDK or an metrics framework built into your application stack, such as Spring Boot Actuator.
+
+After deploying your application, you register it using the `cf` CLI, as described in the documentation: [https://docs.pivotal.io/application-service/metric-registrar/using.html](https://docs.pivotal.io/application-service/metric-registrar/using.html).
+
+## Wavefront Service Broker
+
+To implement a push-based metrics pattern using Wavefront, you can use the Wavefront SDK for your language stack, or you can use a pluggable abstraction like Spring Boot Actuator.
+
+In either case, you will need to provide your application with a URL for a wavefront proxy instance.
+
+The supported way to do this using the Wavefront Nozzle for TAS is to use the Wavefront Service Broker to create a service instance and bind it to your application. You can then push custom metrics to the proxy from your own application code.
+
+## Factors to Consider
+
+### Spring Boot
+
+Spring Boot is commonly used to build apps running on TAS.
+
+Spring Boot Actuator supports both push and pull models, with the only difference typically being configuration changes. In that case, it becomes relatively easy to switch from push to pull or vice-versa.
+
+### Wavefront Proxy Load
+
+All additional custom app metrics increase the load on the Wavefront Proxy. It may be necessary to increase the count of wavefront proxy instances and to deploy an IaaS load balancer to balance traffic across proxy instances.
+
+### Loggregator Firehose and Nozzle Component Load
+
+The Metrics Registrar scrapes metrics endpoints and publishes those metrics through the Loggregator Firehose. Sending additional app metrics through the Loggregator Firehose may require scaling up the Loggregator Firehose. The Firehose can be horizontally scaled to a certain point, but eventually hits a practical limit where additional instances do not increase throughput.
+
+The additional metrics flowing through the firehose will also increase load on two Nozzle components that cannot be horizontally scaled - the Exporters and the Telegraf Agent. In future versions, it may be possible to horizontally scale the exporters. In current versions, you can increase throughput of these components by increasing VM size, but only to a certain point. After this point, you must limit the number of metrics or reduce scrape frequency in order to improve performance.
+
+Most TAS environments will not encounter these scaling bottlenecks. If these scaling issues do occur, this may be a reason to use the push model for sending metrics instead.
+
+### Metric Naming
+
+The Nozzle appends the prefix `tas.` to all metrics, including Metrics Registrar metrics. Metrics collected via the Metrics Registrar will have names starting with `tas.apps`, and will automatically have the "source" tag set to the same foundation name used by platform metrics.
```

Comment:
The Tanzu Observability by Wavefront Nozzle appends the prefix `tas.` to all metrics, including the Metric Registrar metrics. Metrics collected by using the Metric Registrar will have names with the `tas.apps` prefix, and will automatically have the "source" tag set to the same foundation name used by the platform metrics.

For push-based metrics, metric naming and specifying source and point tags must be done by the application sending the metrics or through rewrite rules at the Wavefront proxy level. Some point tags are reserved and cannot be customized by the application. For example, if the application sends tag values for `job`, `origin`, and `source_id`, they will be overwritten while being sent to Tanzu Observability.

Reply:
I made all of the above changes, and one more:
I changed:
``` "source" ```` 
to 
```
`source`
``` 
to match the backticks around `job`, `origin` and `source_id`. 

Is that okay, @Margarita-Staneva?

Reply:
Sounds good, thanks!